File: aqa_python/automation_playwright/pom/editcontact_page.py
from pom.base_page import BasePage

class EditContactPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.page_url = f"{self.config.BASE_URL}/editContact"
        self.selectors = {
            "logout_button": "#logout",
            "edit_form": "#edit-contact",
            "first_name": "#firstName",
            "last_name": "#lastName",
            "email": "#email",
            "birthdate": "#birthdate",
            "phone": "#phone",
            "street1": "#street1",
            "street2": "#street2",
            "city": "#city",
            "state": "#stateProvince",
            "postal_code": "#postalCode",
            "country": "#country",
            "submit_button": "#submit",
            "cancel_button": "#cancel",
            "error_message": "#error"
        }

    def navigate(self):
        self.go_to_url(self.page_url)

    def edit_sample_contact(self, index: int):
        self.page.fill(self.selectors["first_name"], f"edited_firstNamev{index}")
        self.page.fill(self.selectors["last_name"], f"edited_lastNamev{index}")
        self.page.fill(self.selectors["email"], f"edited_emailv{index}@q.com")
        self.page.fill(self.selectors["birthdate"], "1970-01-01")
        self.page.fill(self.selectors["phone"], "001234567890")
        self.page.fill(self.selectors["street1"], f"edited_street1v{index}")
        self.page.fill(self.selectors["street2"], f"edited_street2v{index}")
        self.page.fill(self.selectors["city"], f"edited_cityv{index}")
        self.page.fill(self.selectors["state"], f"edited_statev{index}")
        self.page.fill(self.selectors["postal_code"], "0012345")
        self.page.fill(self.selectors["country"], f"edited_countryv{index}")
        self.page.wait_for_timeout(200)

    def edit_first_name(self, value: str):
        self.page.fill(self.selectors["first_name"], value)
        self.page.wait_for_timeout(200)

    def edit_last_name(self, value: str):
        self.page.fill(self.selectors["last_name"], value)
        self.page.wait_for_timeout(200)

    def edit_email(self, value: str):
        self.page.fill(self.selectors["email"], value)
        self.page.wait_for_timeout(200)

    def edit_birthdate(self, value: str):
        self.page.fill(self.selectors["birthdate"], value)
        self.page.wait_for_timeout(200)

    def edit_phone(self, value: str):
        self.page.fill(self.selectors["phone"], value)
        self.page.wait_for_timeout(200)

    def edit_street1(self, value: str):
        self.page.fill(self.selectors["street1"], value)
        self.page.wait_for_timeout(200)

    def edit_street2(self, value: str):
        self.page.fill(self.selectors["street2"], value)
        self.page.wait_for_timeout(200)

    def edit_city(self, value: str):
        self.page.fill(self.selectors["city"], value)
        self.page.wait_for_timeout(200)

    def edit_state(self, value: str):
        self.page.fill(self.selectors["state"], value)
        self.page.wait_for_timeout(200)

    def edit_postal_code(self, value: str):
        self.page.fill(self.selectors["postal_code"], value)
        self.page.wait_for_timeout(200)

    def edit_country(self, value: str):
        self.page.fill(self.selectors["country"], value)
        self.page.wait_for_timeout(200)

    def click_submit(self):
        self.page.click(self.selectors["submit_button"])
        self.page.wait_for_timeout(200)

    def click_cancel(self):
        self.page.click(self.selectors["cancel_button"])
        self.page.wait_for_timeout(200)

    def click_logout(self):
        self.page.click(self.selectors["logout_button"])
        self.page.wait_for_timeout(200)

    def get_error_message(self) -> str:
        return self.page.text_content(self.selectors["error_message"])

File: aqa_js/automation_playwright/pom/editcontactpage.js
const {Page, expect} = require('@playwright/test');
const BasePage = require('./basepage');

class EditContactPage extends BasePage {
    /**
     * @param {Page} page - Playwright Page object
     */
    constructor(page) {
        super(page);
        this.PageUrl = `${this.config.BASE_URL}/editContact`; // Replace with the actual URL of the signup page
        this.ids = {
            logoutButtonId: "#logout",
            editContactFormId: "#edit-contact",
            firstNameFieldId: "#firstName",
            lastNameFieldId: "#lastName",
            emailFieldId: "#email",
            birthdateFieldId: "#birthdate",
            phoneFieldId: "#phone",
            street1AddressFieldId: "#street1",
            street2AddressFieldId: "#street2",
            cityFieldId: "#city",
            stateFieldId: "#stateProvince",
            postalCodeFieldId: "#postalCode",
            countryFieldId: "#country",
            submitButtonId: '#submit',
            cancelButtonId: '#cancel',
            errorMessageId: '#error'

        };
    }

    async navigate() {
        await super.go_to_url(this.PageUrl);
    }
    async edit_sample_contact(version_index) {
        await this.page.fill(this.ids.firstNameFieldId, `edited_firstNamev${version_index}`);
        await this.page.fill(this.ids.lastNameFieldId, `edited_lastNamev${version_index}`);
        await this.page.fill(this.ids.emailFieldId, `edited_emailv${version_index}@q.com`);
        await this.page.fill(this.ids.birthdateFieldId, '1970-01-01');
        await this.page.fill(this.ids.phoneFieldId, `001234567890`);
        await this.page.fill(this.ids.street1AddressFieldId, `edited_street1v${version_index}`);
        await this.page.fill(this.ids.street2AddressFieldId, `edited_street2v${version_index}`);
        await this.page.fill(this.ids.cityFieldId, `edited_cityv${version_index}`);
        await this.page.fill(this.ids.stateFieldId, `edited_statev${version_index}`);
        await this.page.fill(this.ids.postalCodeFieldId, `0012345`);
        await this.page.fill(this.ids.countryFieldId, `edited_countryv${version_index}`);
    }
    async edit_first_name(firstNameText) {
        await this.page.fill(this.ids.firstNameFieldId, firstNameText);
    }
    async edit_last_name(lastNameText) {
        await this.page.fill(this.ids.lastNameFieldId, lastNameText);
    }
    async edit_email(emailText) {
        await this.page.fill(this.ids.emailFieldId, emailText);
    }
    async edit_birthdate(birthdateText) {
        await this.page.fill(this.ids.birthdateFieldId, birthdateText);
    }
    async edit_phone(phoneText) {
        await this.page.fill(this.ids.phoneFieldId, phoneText);
    }
    async edit_street1(street1Text) {
        await this.page.fill(this.ids.street1AddressFieldId, street1Text);
    }
    async edit_street2(street2Text) {
        await this.page.fill(this.ids.street2AddressFieldId, street2Text);
    }
    async edit_city(cityText) {
        await this.page.fill(this.ids.cityFieldId, cityText);
    }
    async edit_state(stateText) {
        await this.page.fill(this.ids.stateFieldId, stateText);
    }
    async edit_postal_code(postalCodeText) {
        await this.page.fill(this.ids.postalCodeFieldId, postalCodeText);
    }
    async edit_country(countryText) {
        await this.page.fill(this.ids.countryFieldId, countryText);
    }
    async click_submit() {
        await this.page.click(this.ids.submitButtonId);
    }
    async click_cancel() {
        await this.page.click(this.ids.cancelButtonId);
    }
    async click_logout() {
        await this.page.click(this.ids.logoutButtonId);
    }
    async get_error_message() {
        return await this.page.textContent(this.ids.errorMessageId);
    }
}
module.exports = EditContactPage;